### ISSUE TITLE
Polymer build ignore error and does not copy polymer-micro.html ?

### ISSUE THREAD

**Issue by RoXuS** [Polymer/polymer-cli#398]
Hello !

### Description
When we build our application, if we have a component which import an unknow dependency we see this error during the process:
```
error:   Uncaught exception: Error: ENOENT: no such file or directory, open...
```
The exit code is 0 and the `polymer-micro.html` in `bower_components/polymer` is not copied !

#### Steps to Reproduce (version 0.16.0)

1. Choose a component in your bower_components (this component should be call in your code), import a non existing component inside this component.
2. Build: `polymer build`.

#### Expected Results
A return code not equal to 0.

#### Actual Results
During the build, you see the error but the process returns an exit code equal to 0.
In `build/unbundled/bower_components/polymer` the polymer-micro.html is not present.

**Comment by FredKSchott** [comment]
Good catch, thanks for reporting!

**Comment by deltaepsilon** [comment]
I don't know if this is related, but I had a view with a bad import path—../bower_components instead of ../../bower_components—which causes ```polymer build``` to fail.

All of the files get bundled and copied correctly, but I never see the "info: Build complete!" and service-worker.js is not generated. I did some debugging, and it didn't cause either the bundledPhase or the unbundledPhase to throw an ```error``` or ```end``` event.

To reproduce, go to your my-view404.html and add change ../bower_components to /bower_components. Then run ```polymer build``` and watch it fail without throwing any meaningful error.

**Comment by vtellier** [comment]
I have a similar _but sligthly different_ behavior :
- Import a source file with a wrong name
- No error signaled, even with `--verbose`
- The build stops without displaying `'Build complete!'`
- I do have many imports (including `polymer-micro.html`) ; but miss at least the half of my source files.

Here are my log trace, the section that concern the "missnamed" file:
```
debug:   addFile: /home/ubuntu/workspace/myproject/src/myproject-app/list-page.html
debug:   accept: src/myproject-app/shared-styles.html
debug:   dependency is a source file, ignoring... 
{ dep: 'src/myproject-app/shared-styles.html' }
debug:   addFile: /home/ubuntu/workspace/myproject/bower_components/polymerfire/firebase-app.html
```
- list-page.html does exists, it imports the wrong path
- `shared-styles.html` do not exists (the good one is called `shared-style.html` ; without '**s**')
- the path maths my source wildcard in `polymer.json`
- **Version of polymer-cli : 0.16.0**

**Comment by FredKSchott** [comment]
I agree, the silent build problem is really hard to catch when you run into it. We're working to merge in the new analyzer and error handling in asap to fix these issues

**Comment by cordovapolymer** [comment]
Bump the bug is still here.

### Description
Incorrect path in "fragments" breaks build without giving a notice.

### Versions & Environment

- Polymer CLI:
- node: 7.6
- Operating System: Debian stretch

#### Steps to Reproduce


#### Expected Results
An error

#### Actual Results
No error, build stops silently

**Comment by MGRProgramming** [comment]
Any update here how to catch errors - I found this issue as the build step does not produce the service-worker.js file but I cannot find a file that is misspelled.
I updated to latest polymer-cli version
Only odd behaviour is:

> debug:   accept: bower_components/paper-toolbar/paper-toolbar.html
> debug:   new dependency found, pushing into dependency stream... > C:\Users\graupner.martin\Documents\DocsMG\Programming\apps\damawi\bower_components\paper-toolbar\paper-toolbar.html
> Ignoring duplicate element definition: custom-style
> Ignoring duplicate element definition: dom-template
> Ignoring duplicate element definition: dom-repeat
> Ignoring duplicate element definition: array-selector
> Ignoring duplicate element definition: dom-if
> Ignoring duplicate element definition: dom-bind

And I do not get the info built complete
